pkgs/development/compilers/llvm/common/clang-tools/default.nix

```diff
@@ -1,17 +1,22 @@
-{ lib, stdenv, llvmPackages, enableLibcxx ? false }:
-# enableLibcxx will use the c++ headers from clang instead of gcc.
-# This shouldn't have any effect on platforms that use clang as the default compiler already.
-
-let
-  unwrapped = llvmPackages.clang-unwrapped;
-
-in stdenv.mkDerivation {
-  inherit unwrapped;
+{
+  lib,
+  stdenv,
+  clang-unwrapped,
+  clang,
+  libcxxClang,
+  llvm_meta,
+  # enableLibcxx will use the c++ headers from clang instead of gcc.
+  # This shouldn't have any effect on platforms that use clang as the default compiler already.
+  enableLibcxx ? false,
+}:
+
+stdenv.mkDerivation {
+  unwrapped = clang-unwrapped;
 
   pname = "clang-tools";
-  version = lib.getVersion unwrapped;
+  version = lib.getVersion clang-unwrapped;
```

Comment:
nit: the scope that's used in `llvmPackages` already has `version`:

https://github.com/NixOS/nixpkgs/blob/0179f8c547879eb26016fb1dcda2904001487a3d/pkgs/development/compilers/llvm/18/default.nix#L61

You should be able to drop `clang-unwrapped` from this package's attrs (and all of the `llvmPackages_*` set's `default.nix`s and just ask for `version` in this package's attrs.

Reply:
Since `clang-tools` is a wrapper of (a part of) `clang-unwrapped`, the build process would be more straightforward to pass `clang-unwrapped` as a shell variable. It's also okay to string-interpolate it directly.

As for the version, it makes sense to pass it from the llvmPackage an `<pkg>.override` argument.

> You should be able to drop clang-unwrapped from this package's attrs

Are we removing the attribute to prevent people from accessing the `clang-unwrapped` package used via the attribute (or even try to override it via `<pkg>.overrideAttrs`) but encourage accessing/overriding via `llvmPackages` instead?

Reply:
My apologies, I missed that the `$out/bin` symlinks are produced from `clang-unwrapped` and not `clang`/`libcxxClang`.

Please disregard the above; getting `version` from `clang-unwrapped` seems like the sensible thing to do.